benchmarks/source-agilitycms/gatsby-node.js

```diff
@@ -1,57 +1,69 @@
+      sitemapNode: agility.getDynamicPageItemSitemapNode(),
 
-	const resolvers = {
-		//on the 'agilityPost' node type...
-		agilityPost: {
-			//get the sitemap node that represents this item - useful for retrieving the URL for the item
-			sitemapNode: agility.getDynamicPageItemSitemapNode(),
+      //[Not Implemented]
+      //if we had a linked content field for 'author', this is how we'd get the author for this post in a single GraphQl query
+      //linkedContent_agilityAuthor: agility.getLinkedContentItem({ type: 'agilityAuthor', linkedContentFieldName: 'author' })
+    },
 
-			//[Not Implemented]
-			//if we had a linked content field for 'author', this is how we'd get the author for this post in a single GraphQl query
-			//linkedContent_agilityAuthor: agility.getLinkedContentItem({ type: 'agilityAuthor', linkedContentFieldName: 'author' })
-		},
+    //[Not Implemented]
+    //if we had an 'Image Slider' module and it had a list of slides via a linked content field called 'slides', this is how we'd retrieve a list of those slides in a single GraphQL query
+    // agilityImageSlider: {
+    //     linkedContent_agilitySlides: agility.getLinkedContentList({ type: 'agilitySlide', linkedContentFieldName: 'slides' })
+    // }
+  }
+  createResolvers(resolvers)
+}
 
-		//[Not Implemented]
-		//if we had an 'Image Slider' module and it had a list of slides via a linked content field called 'slides', this is how we'd retrieve a list of those slides in a single GraphQL query
-		// agilityImageSlider: {
-		//     linkedContent_agilitySlides: agility.getLinkedContentList({ type: 'agilitySlide', linkedContentFieldName: 'slides' })
-		// }
-	}
-	createResolvers(resolvers)
+function unstable_shouldOnCreateNode({node}) {
+  return (
+    node.properties && node.properties.referenceName.toLowerCase() === `posts`
+  )
 }
 
+exports.unstable_shouldOnCreateNode = unstable_shouldOnCreateNode
+
 exports.onCreateNode = async ({
-	node,
-	actions: { createNode },
-	store,
-	cache,
-	createNodeId,
+  node,
+  actions: { createNode },
+  store,
+  cache,
+  createNodeId,
 }) => {
-
-	if (node.properties
-		&& node.properties.referenceName.toLowerCase() === `posts`) {
-		let field = "image"
-		let imageUrl = node.customFields[field]
-		if (imageUrl) {
-			let fileNode = await createRemoteFileNode({
-				url: imageUrl, // string that points to the URL of the image
-				parentNodeId: node.id, // id of the parent node of the fileNode you are going to create
-				createNode, // helper function in gatsby-node to generate the node
-				createNodeId, // helper function in gatsby-node to generate the node id
-				cache, // Gatsby's cache
-				store, // Gatsby's redux store
-			})
-			// if the file was created, attach the new node to the parent node
-			if (fileNode) {
-				node.customFields[`${field}LocalImg___NODE`] = fileNode.id
-			}
-		}
-	}
-}
\ No newline at end of file
+  if (unstable_shouldOnCreateNode({node})) {
```

Comment:
Curious, why do we have it here? We don't add explicit checks in other benchmarks? Is it for backward compatibility?

Reply:
Yes. But the benchmarks should not need this since they run on the latest version anyways. But I'll remove them from the benchmarks in this PR, anyways.